harness/tests/tensorboard/test_base.py

```diff
@@ -90,3 +94,53 @@ def test_illegal_type() -> None:
         tensorboard.build(
             env.det_cluster_id, env.det_experiment_id, env.det_trial_id, checkpoint_config
         )
+
+
+def test_upload_thread_normal_case() -> None:
+    upload_function = mock.Mock()
+    work_queue = queue.Queue(maxsize=10)
+    upload_thread = tensorboard.base._TensorboardUploadThread(upload_function, work_queue)
+
+    upload_thread.start()
+    work_queue.put([pathlib.Path("test_value/file1.json"), pathlib.Path("test_value/file2.json")])
+    work_queue.put([pathlib.Path("test_value/file3.json")])
+    # Pass in sentinel value to exit thread
+    work_queue.put(None)
+    upload_thread.join()
+
+    assert upload_function.call_count == 2
+
+
+def test_upload_thread_exception_case() -> None:
+    # 1. Set up custom hook to capture threads fail
+    #    with exception. This hook is triggered when
+    #    a thread exits with exception.
+    import threading
+
+    threads_with_exception = set()
+
+    def custom_excepthook(args):
+        thread_name = args.thread.ident
+        threads_with_exception.add(thread_name)
+
+    threading.excepthook = custom_excepthook
```

Comment:
if you keep the test, you'll need a `try`/`finally` to avoid leaking this excepthook into other tests.

Reply:
Good point, let me look into this.